notebooks/09_Monitoring/src/09_01_Iris_Pipeline_Monitoring/model.py

```diff
@@ -0,0 +1,221 @@
+    with file_io.FileIO(roc_file, 'w') as f:
+        df_roc.to_csv(f, columns=['fpr', 'tpr', 'thresholds'], header=False, index=False)
+
+    return cm_file, roc_file, vocab
+
+
+def tf_training(args, X_train, y_train, X_test, y_test):
+    model = tf.keras.Sequential([
+        tf.keras.layers.Dense(10, activation='relu', input_shape=(4,)),  # input shape required
+        tf.keras.layers.Dense(32, activation='relu'),
+        tf.keras.layers.Dense(64, activation='relu'),
+        tf.keras.layers.Dense(128, activation='relu'),
+        tf.keras.layers.Dense(2)
+    ])
+
+    model.compile(optimizer=args.tf_nn_optimizer,
+                  loss=args.tf_nn_loss_func,
+                  metrics=[args.tf_nn_metrics])
+
+    time_hash = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+    log_prefix = args.tf_log_prefix
+    log_dir = os.path.join(log_prefix, time_hash)
+    tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir)
+
+    model.fit(x=X_train,
+              y=y_train,
+              epochs=args.tf_nn_epoch,
+              validation_data=(X_test, y_test),
+              callbacks=[tensorboard_callback])
+
+    return model, log_dir, time_hash
+
+
+def upload_s3(args, cm_file, roc_file ,log_dir, time_hash):
+    AWS_REGION = args.aws_region
+    s3_client = boto3.client('s3', region_name=AWS_REGION)
+    bucket = args.s3_bucket
+    s3_file_path_prefix = args.s3_file_path_prefix
+
+    try:
+        # upload cm file to S3
+        cm_object_name = 'confusion_matrix.csv'
+        s3_cm_file = os.path.join('s3://', bucket, s3_file_path_prefix, cm_object_name)
+        s3_client.upload_file(cm_file, bucket, cm_object_name)
+
+        roc_object_name = 'roc.csv'
+        s3_roc_file = os.path.join('s3://', bucket, s3_file_path_prefix, roc_object_name)
+        s3_client.upload_file(roc_file, bucket, roc_object_name)
+
+        # upload tb log dir to S3
+        tb_object_name = 'tb-logs'
+        s3_tb_file = os.path.join('s3://', bucket, s3_file_path_prefix, tb_object_name)
+        for path, subdirs, files in os.walk(log_dir):
+            path = path.replace("\\", "/")
+            directory_name = path.replace(log_dir, "")
+            for file in files:
+                if directory_name and directory_name[0] == '/':
+                    directory_name = directory_name[1:]
+                s3_client.upload_file(os.path.join(path, file), bucket,
+                                      os.path.join(s3_file_path_prefix, 'tb-logs', time_hash, directory_name, file))
+    except ClientError as e:
+        logging.info("ERROR IN S3 UPLOADING!!!!!!")
+        logging.ERROR(e)
+
+    logging.info("S3 bucket path is: s3://{}/iris-example".format(bucket))
+
+    return s3_cm_file, s3_roc_file, s3_tb_file
+
+
+def write_metadata(args, s3_cm_file, s3_roc_file, s3_tb_file, vocab):
+    metadata = {
+        'outputs': [
+            {
+                'source': 's3://{}/iris-example/README.md'.format(args.s3_bucket),
```

Comment:
I am still not sure where does README.md come from , I didn't see you upload a README.md to the bucket?  Maybe let's just use fixed value `https://github.com/aws-samples/eks-kubeflow-workshop/blob/master/README.md`?

Reply:
It's fixed in the new commit, I should have fixed it yesterday, not sure why the commit I pushed doesn't match with my local commit.